### PR TITLE
Feature/enable https with certbot

### PR DESCRIPTION
**Pull request by frasercrichton**
Ansible script to install a certificate on the server. Have treated this as a separate task as it simplifies spinning up ephemeral instances of cloud VM's.

### CODE REVIEW

File: deploy_timemap/templates/nginx.conf.j2 [hunk 1]
```diff
@@ -7,4 +7,7 @@ server {
 	location / {
 		return 403;
 	}
+
+	# enter certbot block below
```
Comment (breezykermo) [inline deploy_timemap/templates/nginx.conf.j2:11]: wow this is also a much nicer way of inserting into a particular place. I've noticed that the tasks in 'nginx.yaml' sometimes incorrectly insert lines at the end of the configuration- I'll add an issue to update them to work like this!

File: provision/gcp-instance/gcp-cors.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[
+  {
+    "origin": ["*"],
+    "responseHeader": ["Content-Type"],
+    "method": ["GET"],
+    "maxAgeSeconds": 3600
+  }
+]
```
Comment (breezykermo) [inline provision/gcp-instance/gcp-cors.json:8]: is this code related to certbot provisioning? Or is it a fix of something else?

File: deploy_timemap/tasks/certbot.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+- name: update apt {{ letsencrypt_email }}
+  apt:
+    upgrade: yes
+    update_cache: yes
+    cache_valid_time: 3600
+
+- name: Install certbot
+  package:
+    name: ['certbot','python-certbot-nginx']
```
[the remark this answers is not in the document]
Comment (frasercrichton) [reply deploy_timemap/tasks/certbot.yml:9]: Tricky, I don't think it does but there was a bit of oddness. Let me delete and re-run on  GCP and if it's okay without I'll delete.

File: deploy_timemap/tasks/certificate.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+- name: Force generation of a new certificate
+  shell: >
+    certbot certonly --force-renewal --nginx --email '{{ letsencrypt_email }}'
+    --agree-tos -d '{{ domain_name }}'
+    {% if certbot_nginx_cert_name is defined %}
+    --cert-name '{{ certbot_nginx_cert_name }}'
+    {% endif %}
+    {% if letsencrypt_staging %} --staging {% endif %}
+  when: letsencrypt_cert.stat.exists and certbot_force_update is defined
+  notify: reload nginx
```
[the remark this answers is not in the document]
Comment (frasercrichton) [reply deploy_timemap/tasks/certificate.yml:25]: Now I can't take credit for that ;) but good that it's elegant.

File: deploy_timemap/templates/nginx.conf.j2 (hunk 1 above)
Comment (frasercrichton) [reply deploy_timemap/templates/nginx.conf.j2:11]: Sweet! Yeah, the config file is really tricky as it feels pretty fragile if it's already there.

File: provision/gcp-instance/gcp-cors.json (hunk 2 above)
Comment (frasercrichton) [reply provision/gcp-instance/gcp-cors.json:8]: No, it's not. It's really part of the other PR and making storage buckets accessible to Timemap. Will remove it and add there.

File: deploy_timemap/tasks/certbot.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (frasercrichton) [reply deploy_timemap/tasks/certbot.yml:9]: Looks like it needs python-certbot-nginx to run the full install. Not so sure about certbot alone. What do you reckon?
